manticore/platforms/linux.py

```diff
@@ -986,20 +1000,18 @@ def sys_lseek(self, fd, offset, whence):
 
         :return: 0 (Success), or EBADF (fd is not a valid file descriptor or is not open)
 
-         '''
-        if not self._is_open(fd):
-            logger.info("LSEEK: Not valid file descriptor on lseek. Returning EBADF")
-            return -errno.EBADF
-
-        if isinstance(self.files[fd], Socket):
-            logger.info("LSEEK: Not valid file descriptor on lseek. Fd not seekable. Returning EBADF")
+        '''
+        try:
+            # Read the data and put it in memory
+            if not self._is_open(fd):
+                raise BadFd()
+            self.files[fd].seek(offset)
+        except BadFd:
```

Comment:
maybe catch both IndexError and AttributeError (None.seek(offset))? or introduce a helper function for accessing an open fd that takes care of indexing into `self.files` but raises a exception if there was None there, or it was out of range